pkg/autostart/autostart.go

```diff
@@ -0,0 +1,119 @@
+// Package autostart manage start at login unit files for darwin/linux
+package autostart
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/lima-vm/lima/pkg/textutil"
+)
+
+//go:embed lima-vm@INSTANCE.service
+var systemdTemplate string
+
+//go:embed io.lima-vm.autostart.INSTANCE.plist
+var launchdTemplate string
+
+// CreateStartAtLoginEntry respect host OS arch and create unit file
+func CreateStartAtLoginEntry(hostOS, instName, workDir string) error {
+	unitPath := GetStartAtLoginEntryPath(hostOS, instName)
+	if _, err := os.Stat(unitPath); err != nil && !errors.Is(err, os.ErrNotExist) && !errors.Is(err, os.ErrExist) {
+		return err
+	}
+	tmpl, err := renderTemplate(hostOS, instName, workDir, os.Executable)
+	if err != nil {
+		return err
+	}
+	if err := os.MkdirAll(filepath.Dir(unitPath), os.ModePerm); err != nil {
+		return err
+	}
+	if err := os.WriteFile(unitPath, tmpl, 0o644); err != nil {
+		return err
+	}
+	return enableDisableService("enable", hostOS, GetStartAtLoginEntryPath(hostOS, instName))
+}
+
+// DeleteStartAtLoginEntry respect host OS arch and delete unit file
+// return true, nil if unit file has been deleted
+func DeleteStartAtLoginEntry(hostOS, instName string) (bool, error) {
+	unitPath := GetStartAtLoginEntryPath(hostOS, instName)
+	if _, err := os.Stat(unitPath); err != nil {
+		return false, err
+	}
+	if err := enableDisableService("disable", hostOS, GetStartAtLoginEntryPath(hostOS, instName)); err != nil {
+		return false, err
+	}
+	if err := os.Remove(unitPath); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// GetFilePath returns the path to autostart file with respect of host
```

Comment:
```suggestion
// GetStartAtLoginEntryPath returns the path to autostart file with respect of host
```

Given that we call this the "autostart file", maybe this function should be called `GetStartAtLoginFileName`? I'm not sure what `EntryPath` means.

Reply:
> I'm not sure what EntryPath means.

It means an entry of some file. `GetFilePath` doesn't say what kind of file path this function returns.